layer/projects/progress_tracker_updater.py

```diff
@@ -116,14 +117,23 @@ def _handle_task_succeeded(self, task: PBTask) -> None:
         task_name = self._get_name_from_task_id(task_id)
         task_type = task.type
         if task_type == PBTask.TYPE_DATASET_BUILD:
+            dataset_build_id = uuid.UUID(
+                self.run_metadata[(task_type, task_id, "build-id")]
+            )
```

Comment:
can we have a `run_metadata.get_dataset_build_id()` instead?

or are we too close to the API here? from the name `progress_tracker_updater` it seems we're dealing with protos way to much in this class